docs/marketplace/add_to_marketplace/add_item_by_type/add_application.md

```diff
@@ -0,0 +1,760 @@
+
+Currently, applications support 4 types of resources:
+
+- Services (Plugins, Templates, Examples)
+- Endpoints
+- CRUD collections
+- Public Variables
+
+Therefore an application will include a `resources` field with the following structure:
+
+```json
+{
+  "services": {...},
+  "endpoints": {...},
+  "collections": {...},
+  "unsecretedVariables": {...},
+  "listeners": {...},
+}
+```
+
+:::info
+Only the `services` field must be configured for an application configuration to be valid.
+:::
+
+### Configure Services
+
+The `service` object has the same structure found in the Plugins, Templates and Examples, as explained in the [dedicated paragraph](/marketplace/add_to_marketplace/contributing_overview.md#common-to-microservice-items-plugins-templates-examples).
+
+Thus, if you plan to add a plugin, a template, or an example to your application you can follow the same configuration. Remember to add general details to every microservice, such as name and description.
+
+:::tip
+
+If you need to include a Plugin, Template or Example that is already on the Marketplace within your application, you can simply download it with `miactl get` and copy its services to your application.
+
+:::
+
+### Configure Endpoints
+
+The `endpoints` field of an application is an object composed of properties named as the endpoints composing it.
+
+Here below are listed all the properties that must be provided for each endpoint:
+
+- **`defaultBasePath`**: the base path of the endpoint
+- **`defaultPathRewrite`**: the default path rewrite of the endpoint
+- **`description`**: a brief description (10 to 20 words) regarding the endpoint functionalities
+- **`type`**: a label to identify the type of endpoint:
+  - **`custom`**: linked to a Microservice
+  - **`crud`**: linked to a CRUD collection
+  - **`view`**: linked to a MongoDB view
+  - **`external`**: linked to an External Proxy
+  - **`cronjob`**: linked to a CronJob
+  - **`cross-project`**: linked to a cross-project proxy
+  - **`fast-data-projection`**: linked to a Fast Data projection
+  - **`fast-data-single-view`**: linked to a Fast Data single view
+- **`tags`**: array of strings to arbitrarily catalog the endpoint in the Console
+- **`public`**: whether the endpoint should require authentication
+- **`showInDocumentation`**: whether the endpoint should appear in the API Portal section
+- **`secreted`**: whether the endpoint should require an API key
+- **`service`**: present only in custom type endpoints, identifies the microservice the endpoint is linked to
+- **`collectionId`**: present only in CRUD type endpoints, identifies the collection the endpoint is linked to
+- **`pathName`**: present only in CRUD type endpoints, identifies the path name of the collection di endpoint is linked to
+  
+:::caution
+When you add an endpoint to an application make sure the resource the endpoint is linked to (microservice, collection, etc.) is also present in the application configuration.
```

Comment:
Non capisco cosa si volesse intendere con questa frase. Quando crei una applicazione che i suoi endpoint essi vengono creati e linkati al servizio che va a crearti

Reply:
Secondo me intende che non puoi mettere endpoint che si riferiscono a risorse non incluse nel `resources` dell'applicazione stessa. 

In realtà mi torna come cosa, forse si può spiegare meglio. A te torna?